app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,126 @@
+samson.controller('ProjectRolesCtrl', function ($scope, $element, $filter, projectRolesService, messageCenterService) {
+    $scope.project_role = {};
+    $scope.roles = [];
+
+    $scope.initModel = function () {
+        $scope.project_role = {
+            id: getUniqueId(),
+            user_id: getUserId(),
+            project_id: getProjectId(),
+            role_id: getRoleId()
+        };
+
+        loadProjectRoles();
+    };
+
+    $scope.$watch('project_role.role_id', function (new_role_value, old_role_value) {
+        if (new_role_value !== old_role_value) {
+            if (exists($scope.project_role)) {
+                return updateProjectRole($scope.project_role, new_role_value);
+            }
+            else {
+                return createProjectRole($scope.project_role, new_role_value);
+            }
+        }
+    });
+
+    function loadProjectRoles() {
+        projectRolesService.loadProjectRoles().then(
+            function (response) {
+                $scope.roles = response.data;
+            }
+        );
+    }
+
+    function exists(project_role) {
+        return project_role.id;
+    }
+
+    function createProjectRole(project_role) {
+        projectRolesService.createProjectRole(project_role).then(
+            function (response) {
+                var message = 'User ' + getUserName() + ' has been granted the role ' + roleNameFor(project_role.role_id) + ' for project ' + getProjectName();
+                showSuccessMessage(message);
+                project_role.id = response.data.project_role.id;
+            },
+            function () {
+                var message = "Failed to assign role '" + roleNameFor(project_role.role_id) + "' to User " + getUserName() + " on project " + getProjectName();
+                showErrorMessage(message);
+            }
+        );
+    }
+
+    function updateProjectRole(project_role) {
+        projectRolesService.updateProjectRole(project_role).then(
+            function () {
+                var message = 'User ' + getUserName() + ' has been granted the role ' + roleNameFor(project_role.role_id) + ' for project ' + getProjectName();
+                showSuccessMessage(message);
+            },
+            function () {
+                var message = "Failed to assign role '" + roleNameFor(project_role.role_id) + "' to User " + getUserName() + " on project " + getProjectName();
+                showErrorMessage(message);
+            }
+        );
+    }
+
+    function getUniqueId()
+    {
+        return getAttributeAsInt('data-id');
+    }
```

Comment:
Yeah... I probably went too far with the getters. Was just trying to keep the actual code simpler and cleaner.

I removed all getters except for `getUserName()` and `getProjectName()`. Each one of this methods is used in several places, so, I think it's cleaner to keep the logic responsible for getting the corresponding attribute values in a separate utility method rather than having it scattered across the code.

Are you guys ok with that?